WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreOnboardingView.swift

```diff
@@ -15,43 +56,80 @@ struct StoreOnboardingView: View {
-                                   numberOfTasksCompleted: viewModel.numberOfTasksCompleted)
-
-            // Task list
-            VStack(alignment: .leading, spacing: Layout.verticalSpacingBetweenTasks) {
-                ForEach(Array(viewModel.taskViewModels.enumerated()), id: \.offset) { index, taskViewModel in
-                    let isLastTask = index == viewModel.taskViewModels.count - 1
-
-                    StoreOnboardingTaskView(viewModel: taskViewModel,
-                                            showDivider: !isLastTask) { task in
-                        taskTapped(task)
+        if viewModel.isExpanded {
+            ScrollView {
+                content
+            }
+        } else {
+            content
+        }
+    }
+
+    private var content: some View {
+        VStack {
+            Color(uiColor: .listBackground)
+                .frame(height: Layout.VerticalSpacing.collapsedMode)
+                .renderedIf(!viewModel.isExpanded)
+
+            let verticalSpacing = viewModel.isExpanded ? Layout.VerticalSpacing.expandedMode : Layout.VerticalSpacing.collapsedMode
+            VStack(alignment: viewModel.isExpanded ? .center : .leading, spacing: verticalSpacing) {
+                // Progress view
+                StoreSetupProgressView(isExpanded: viewModel.isExpanded,
+                                       totalNumberOfTasks: viewModel.taskViewModels.count,
+                                       numberOfTasksCompleted: viewModel.numberOfTasksCompleted)
+
+                // Task list
+                VStack(alignment: .leading, spacing: Layout.verticalSpacingBetweenTasks) {
+                    ForEach(viewModel.tasksForDisplay) { taskViewModel in
+                        let isLastTask = taskViewModel == viewModel.tasksForDisplay.last
+
+                        StoreOnboardingTaskView(viewModel: taskViewModel,
+                                                showDivider: !isLastTask) { task in
+                            taskTapped(task)
+                        }
                     }
                 }
-            }
 
-            // View all button
-            Button {
-                viewAllTapped?()
-            } label: {
-                Text(String(format: Localization.viewAll, viewModel.taskViewModels.count))
-                    .fontWeight(.semibold)
-                    .foregroundColor(.init(uiColor: .accent))
-                    .subheadlineStyle()
+                // View all button
+                viewAllButton(action: viewAllTapped, text: String(format: Localization.viewAll, viewModel.taskViewModels.count))
+                    .renderedIf(!viewModel.isExpanded)
+
+                Spacer()
+                    .renderedIf(viewModel.isExpanded)
             }
-            .renderedIf(!viewModel.isExpanded)
+            .padding(insets: Layout.insets)
+            .background(Color(uiColor: .listForeground(modal: false)))
```

Comment:
Seems like this background causes the expanded mode to look odd with different colors:

<img src="https://user-images.githubusercontent.com/5533851/222040714-dacad4ff-b518-4673-88af-70b96895c903.png" width=320 />

Let's apply this background color only for the collapsed mode:

```suggestion
            .background(Color(uiColor: .listForeground(modal: false)).renderedIf(!viewModel.expandedMode))
```

Reply:
I missed checking this. Done in https://github.com/woocommerce/woocommerce-ios/pull/8991/commits/73413fbbbde53c24d7cb9ae1cce0030c6a344f48